src/master-playlist-controller.js

```diff
@@ -1161,22 +1161,20 @@ export class MasterPlaylistController extends videojs.EventTarget {
     // audio). In the future, we may parse codec info from the segments, but for now, we
     // rely on the manifest or defaults, so don't have to wait for the alt audio segment.
 
-    const hasVideo = this.mainSegmentLoader_.startingMedia_.hasVideo;
-    const hasAudio = this.mainSegmentLoader_.startingMedia_.hasAudio ||
-      // alt audio always has audio
-      this.mediaTypes_.AUDIO.activePlaylistLoader;
+    const startingMedia = this.mainSegmentLoader_.startingMedia_;
+    const {hasVideo, videoCodec} = startingMedia;
     const media = this.masterPlaylistLoader_.media();
+    // alt audio always has audio
+    // TODO: should we wait for `trackinfo` from audioSegmentLoader?
```

Comment:
I think just having the codec info is enough to create the source buffers

Reply:
The problem is when we have a video segment loader that just has video and and audio segment loader that just has audio. We want to create two source buffers, but right now we will only create one.

Reply:
hmm, is this different to the pre-TB4A behavior?

Reply:
previously we just use the codec from the playlist, now we use the codec that we parse from the mp4 init segment, if available.

Reply:
ah, yeah, but I mean why would we need to wait for `trackinfo` from the audioSegmentLoader?

Reply:
If we don't have codec info in the playlist and we have separate audio/video content The first `trackinfo` will fire for either audio or video. Which will mean that we create one sourceBuffer for audio or video. Then the other trackinfo will be ignored because sourceBuffers were already created.

Reply:
I see. Is that a problem with the current implementation? and I assume looking at the tracks defined in the init segment will help with this?

Reply:
This was a problem with the current implementation yeah. Looking at the tracks codecs fixes this, but we have to wait for two `trackinfo` events. I updated the code to do that, so we should be good to go.